tools/bridge/bridge/transfer_fetcher.py

```diff
@@ -1,25 +1,135 @@
 import logging
-
 import gevent
-from gevent.queue import Queue
 
+from gevent.queue import Queue
 from web3 import Web3
 
+MINIMAL_TOKEN_ABI = [
+    {
+        "anonymous": False,
+        "inputs": [
+            {"indexed": True, "name": "from", "type": "address"},
+            {"indexed": True, "name": "to", "type": "address"},
+            {"indexed": False, "name": "value", "type": "uint256"},
+        ],
+        "name": "Transfer",
+        "type": "event",
+    }
+]
+
+TRANSFER_EVENT_SIGNATURE_HASH = Web3.keccak(text="Transfer(address,address,uint256)")
+
+
+class TransferFetcher:
+    def __init__(
+        self,
+        *,
+        queue: Queue,
+        w3_foreign: Web3,
+        token_contract_address: str,
+        foreign_bridge_contract_address: str,
+        max_reorg_depth: int,
+        event_filter_fetch_limit: int,
+    ):
+        token_contract_code = w3_foreign.eth.getCode(token_contract_address)
+
+        if not token_contract_code:
+            raise ValueError(
+                f"The given token contract address {token_contract_address} "
+                "does not point to a contract!"
+            )
+
+        if TRANSFER_EVENT_SIGNATURE_HASH not in token_contract_code:
+            raise ValueError(
+                f"The token contract at the given address {token_contract_address}"
+                "does not have the (correct) 'Transfer' event!"
+            )
+
+        foreign_bridge_contract_code = w3_foreign.eth.getCode(
+            foreign_bridge_contract_address
+        )
+
+        if not foreign_bridge_contract_code:
+            raise ValueError(
+                f"The given foreign bridge contract address {foreign_bridge_contract_address} "
+                "does not point to a contract!"
+            )
+
+        self._logger = logging.getLogger("bridge.transfer_fetcher")
```

Comment:
I only recently learned this: The underscore prefix signals "private", not "protected". I guess almost all instance variables should rather be protected than private, so I think we should drop pretty much all of those underscores here. Not sure what the rest of the team thinks about this though.

Reply:
Yes, I'd get rid of the underscores. The point is that it's not only a signal for people who use it, but also for people who subclass it.

Reply:
:heavy_check_mark: